src/main/java/com/github/cameltooling/lsp/internal/parser/CamelKYamlDSLParser.java

```diff
@@ -0,0 +1,84 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.github.cameltooling.lsp.internal.parser;
+
+import org.eclipse.lsp4j.Position;
+import org.eclipse.lsp4j.TextDocumentItem;
+
+import com.github.cameltooling.lsp.internal.instancemodel.CamelURIInstance;
+import com.github.cameltooling.lsp.internal.instancemodel.YamlDSLModelHelper;
+
+/**
+ * @author Lars Heinemann
+ */
+public class CamelKYamlDSLParser extends ParserFileHelper {
+	
+	public static final String STEP_KEY = "steps:";
+	public static final String URI_KEY = "uri:";
+	public static final String REST_KEY = "rest:";
+	public static final String FROM_KEY = "- from:";
+	public static final String TO_KEY = "- to:";
+
+	@Override
+	public String getCamelComponentUri(String line, int characterPosition) {
+		String camelComponentURI = null;
+		String trLine = line.trim();
+		if (trLine.startsWith(URI_KEY) && URI_KEY.length() < characterPosition) {
+			camelComponentURI = trLine.substring(trLine.indexOf(URI_KEY)+ URI_KEY.length()).trim().replaceAll("\"", "");
+		} else if(trLine.startsWith(FROM_KEY) && trLine.indexOf('"') >= trLine.indexOf(':') + 1) {
```

Comment:
if I understand well the expression, it expects that values are contained inside double-quotes. In fact, it can be single-quoted or having no quote at all. 
see https://yaml.org/spec/1.2/spec.html#id2786942

maybe we can support the tree relatively easily? at least the single-quote?
Otherwise, please create another issue, and the best will be to use a real yaml parser, we don't really want to bother with all these details :s

Reply:
will check that. Thanks for the pointer.

Reply:
created https://github.com/camel-tooling/camel-language-server/issues/292 to track the issue.